RAPIDS.cmake

```diff
@@ -1,5 +1,5 @@
 #=============================================================================
```

Comment:
You shouldn't need to modify the `RAPIDS.cmake` to test your PR. The testing infrastructure will do this automatically since we don't need to fetch rapids-cmake.

For downstream PR's to verify these changes what you can do is:
```
if(NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/RMM_RAPIDS.cmake)
  set(rapids-cmake-repo bdice/rapids-cmake)
  set(rapids-cmake-branch cccl-update-2.1.0)
  file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-23.06/RAPIDS.cmake
endif()
include(${CMAKE_CURRENT_BINARY_DIR}/RMM_RAPIDS.cmake)
```
That will cause rmm to use the `23.06` version of the `RAPIDS.cmake` file, but that file will download your custom fork/branch of rapids-cmake ( weird but works ).

Reply:
I looked at this and recognized that your suggestion was the intended path — but I was really eager to have a simple sed replacement for the downstream repo tests. I saw that this method of testing (modifying the upstream repo default) let me get by with fewer lines of downstream changes. As a result, I can easily use r3 (an automation tool from @ajschmidt8) to open up testing PR en masse for all repos. Is there anything that could go wrong with my approach? I would revert my changes in this PR before merging, of course.

Reply:
The change in `rmm` doesn't work on local builds when the build directory already exists. The `RMM_RAPIDS.cmake` already exists in the build directory so the new version will never be downloaded.

Reply:
I see. For now, I'm only testing in CI, so I won't hit that case. I can try to build a more complex sed command to do this the other way.

Reply:
I found a better sed command so I'll be using this for subsequent testing PRs.

```bash
find . -name "CMakeLists.txt" -type f -exec sed -i 's/\(^[[:space:]]*\)\(.*branch-23.06\/RAPIDS\.cmake.*\)/\1set(rapids-cmake-repo bdice\/rapids-cmake)\n\1set(rapids-cmake-branch cccl-update-2.1.0)\n\1\2/g' {} \;
```